CRAYSAT-1931: DST Alert to Update SAT in ShastaRelm-release

annapoorna-s-alt opened the pull request:
## Summary and Scope

_Update the [hpc-shastarelm-release](https://github.hpe.com/hpe/hpc-shastarelm-release) vendored release library in sat-product-stream._

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CRAYSAT-1931](https://jira-pro.it.hpe.com:8443/browse/CRAYSAT-1931)

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_


## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

annapoorna-s-alt committed:
CRAYSAT-1931: DST Alert: SAT ShastaRelm-release Update

haasken-hpe commented:
I took a look at the build failure, and it looks like there was a change in the vendored release library with respect to how Artifactory credentials are passed in to Docker when downloading artifacts.

Here's how I went about finding the cause of the problem. First, I checked out the hpc-shastarelm-release repo and did a `git blame` on the `lib/release.sh` file. Since our build is failing in the call to `rpm-sync`, I looked for any recent changes to that function. I found several, and this one in particular affected how credentials are passed: https://github.hpe.com/hpe/hpc-shastarelm-release/commit/4676cb561d3ded26c8e17f19101a4afa155064e9

This commit is one that was pulled in when you did the `git vendor update`.

This commit changed the `rpm-sync` and `helm-sync` functions so that they take pass in credentials via an environment variable instead of a file, and the source of the credentials is now `REPOCREDSVARNAME` instead of `REPOCREDSPATH` and `REPOCREDSFILENAME`. I can see that we are still using the `REPOCREDSPATH` and `REPOCREDSFILENAME` here: https://github.com/Cray-HPE/sat-product-stream/blob/1e8179b7151e37155340f8fdadd8d3b7a16d0f16/release.sh#L121-L133

I took a look at the csm repo to see how they are defining the new `REPOCREDSVAR` and `REPOCREDSVARNAME`, and I found it here: https://github.com/Cray-HPE/csm/blob/bf99907742ade179b06593746b93214db5b412a9/common.sh#L40-L43

So all you should need to do to get this working is replace the code we have in `release.sh` that sets `REPOCREDSPATH` and `REPOCREDSFILENAME` with the couple lines from the csm repo that set `REPOCREDSVAR` and REPOCREDSVARNAME`. It's unclear to me whether the `REPO_CREDS_DOCKER_OPTIONS` and `REPO_CREDS_RPMSYNC_OPTIONS` variables also need to be set because it looks like those would be set in `rpm-sync` anyway. You can try without those two lines and see if it works.

haasken-hpe committed:
Squashed 'vendor/github.hpe.com/hpe/hpc-shastarelm-release/' changes from 4d87937..493a6ff

493a6ff DST-13441: add podman_run_flags to some docker calls that use -v volume mapping (#59)
c3d63e8 DST-13099 add auto detect container software on builder (#58)
b7e73d0 Bump inspect/copy timeouts to 600s
543ed95 Merge pull request #57 from hpe/feature/install-retry-cmd
4b003dd CASMINST-6842 Add retry logic to setup-nexus.sh script
3a1de81 Merge pull request #56 from hpe/feature/aarch64-images
876626c CASMINST-6840 Support multi-platform docker images
12dd965 Merge pull request #55 from hpe/feature/ALT-1096
e0adb99 ALT-1096 Remove hardcoded Nexus username/password
ea40795 Merge pull request #54 from hpe/feature/algol60-auth
c32f52f CASMINST-6835 Use auth in vendor-install-deps
80bf480 Merge pull request #53 from hpe/feature/include-rpm-tools
2bdf2bf Add support for rpm-tools
f49f495 DST-11708: add set -e back to the end of extract-from-container function, if it was in effect before the call (#51)
f37f6ce Merge pull request #52 from hpe/feature/skopeo-1.13
4aa958c CASMINST-6629 Use skopeo 1.13.2
4525f48 Merge pull request #50 from hpe/feature/packaging-tools-0.13.0
f244c64 CASMINST-6507 use packaging-tools 0.13.0 with new rpm-index functionality
efa96c7 Merge pull request #49 from hpe/casminst-6464
bf3a1c0 changes to avoid unbound variable error related to ARTIFACTORY_USER
5619406 Merge pull request #48 from hpe/casminst-6457
47a3595 update test for non-existent REPOCREDSVARNAME
8f3c904 prevent unbound variable error for REPOCREDSVARNAME
d124d69 Merge pull request #47 from hpe/feature/packaging-tools-0.12.7
c5dcbdf CASMINST-6382 Use packaging-tools 0.12.7 with rpm-sync exception hadlingfix
2e1b17f Merge pull request #46 from hpe/CASMINST-6264
7ac7786 bump version of packaging-tools image to v0.12.6
7c2505a DST-11222: add function rpm-sync-src-latest which adds the --src arg to arti-helper so that src rpms will be additionally copied (#45)
23ea106 bump packaging-tools version to 0.12.5
fa091ee Merge pull request #44 from hpe/CASMINST-5679-update-cray-nls-version
645404f CASMINST-5679: Update cray-nls version
260e3c9 Merge pull request #43 from hpe/CASMINST-5876-update-iuf-manifest-validation
c704c94 CASMINST-5876: Update version of cray-nls used to validate IUF manifest
bb5f8ca bump version of packaging-tools image to 0.12.4
40e7d62 Merge pull request #42 from hpe/casminst-5655
4676cb5 use environment variable to pass creds instead of file
7bef9ed fix typo
834dfaa changes to make repo creds JSON file available to helm-sync

git-subtree-dir: vendor/github.hpe.com/hpe/hpc-shastarelm-release
git-subtree-split: 493a6ffa0063f976e65ae8a00030e7d2c142116e

haasken-hpe committed:
Update "release" from "git@github.hpe.com:hpe/hpc-shastarelm-release.git@master"

git-vendor-name: release
git-vendor-dir: vendor/github.hpe.com/hpe/hpc-shastarelm-release
git-vendor-repository: git@github.hpe.com:hpe/hpc-shastarelm-release.git
git-vendor-ref: master
